src/Stratis.Bitcoin/P2P/Peer/NetworkPeerCollection.cs

```diff
@@ -96,17 +97,17 @@ public INetworkPeer FindById(int peerId)
             return this.networkPeers.FirstOrDefault(n => n.Connection.Id == peerId);
         }
 
-        public List<INetworkPeer> FindByIp(IPAddress ip)
+        public List<INetworkPeer> FindByIp(IPAddress ip, int? port = null)
         {
-            ip = ip.EnsureIPv6();
-            return this.networkPeers.Where(n => n.MatchRemoteIPAddress(ip)).ToList();
+            if (!ip.IsRoutable(false))
```

Comment:
Why IsRoutable check here? Is this because we check local IP addresses differently to remote IPs?

Reply:
Yes. See above.